src/services/appointment.service.ts

```diff
@@ -108,10 +108,33 @@ export default class AppointmentService implements OnModuleInit {
     updateDto: AppointmentUpdateDto,
   ): Promise<Appointment> {
     const appointment = await this.searchForId(id);
+
+    if (updateDto.customer_name) {
+      appointment.customer.name = updateDto.customer_name;
+      await this.customerRepository.save(appointment.customer);
+    }
+
+    if (updateDto.treatments_id && Array.isArray(updateDto.treatments_id)) {
+      const treatments = await this.treatmentRepository.find({
+        where: { id: In(updateDto.treatments_id) },
+      });
+
+      if (treatments.length !== updateDto.treatments_id.length) {
+        throw new NotFoundException(`Some treatments not found`);
```

Comment:
I think it would be more useful if you provide the list of treatments that were not found:

```suggestion
        throw new NotFoundException(`Treatments: #{treatments ids} do not exists`);
```

Reply:
Solved